Mathlib/Data/PNat/Basic.lean

```diff
@@ -335,6 +335,23 @@ theorem sub_coe (a b : ℕ+) : ((a - b : ℕ+) : ℕ) = ite (b < a) (a - b : ℕ
     rfl
 #align pnat.sub_coe PNat.sub_coe
 
+theorem sub_le {a b : ℕ+} : a - b ≤ a := by
+  rw [← coe_le_coe, sub_coe]
+  split_ifs with h
+  · exact Nat.sub_le a b
+  · exact a.2
+
+theorem le_sub_one_of_lt {a b : ℕ+} (hab: a < b) : a ≤ b - (1 : ℕ+) := by
+  rw [← coe_le_coe, PNat.sub_coe]
+  split_ifs with h
+  · apply Nat.le_pred_of_lt hab
+  · apply not_lt.mp at h
+    simp_all only [le_one_iff, not_lt_one]
```

Comment:
This can be done slightly quicker by using some neat shortcuts: `LT.lt.le` turns `a < b` into `a ≤ b`, `LE.le.trans` is a short way of writing transitivity, and `not_lt.mp` is defined to equal `le_of_not_lt`. Finally, dot notation means we can turn `LT.lt.le hab` into `hab.lt` and similarly for `.trans`. So we get:
```suggestion
  · exact hab.le.trans (le_of_not_lt h)
```